### PR TITLE
add data and time to measurement file

### LINKED ISSUE #384: [FEATURE] Capture measurement date and time in metadata
**Is your feature request related to a problem? Please describe.**
Measurement file (and comparison files too) should contain date & time when measurement was conducted. This data should be saved in the metadata section. This information should be displayed in the comparison report.

**Describe the solution you'd like**
Capture current date and time at the time of measurement and save it to measurement file. Use ISO format for storing that date.

**Describe alternatives you've considered**
N/A

**Additional context**
Adding capture date will be useful, especially for baseline reports.

### PR DESCRIPTION
**Pull request by josemak25**
### Summary
This PR adds date and time to measurement when running `yarn reassure` 

Resolves #384

### Test plan
All checks should pass.

### CODE REVIEW

File: packages/reassure-cli/src/commands/measure.ts [hunk 1]
```diff
@@ -22,10 +22,13 @@ export async function run(options: MeasureOptions) {
   configureLoggerOptions(options);
 
   const measurementType = options.baseline ? 'Baseline' : 'Current';
+  const date = new Date();
 
   const metadata: PerformanceMetadata = {
+    date: date.toISOString().substring(0, 10),
```
Comment (mdjastrzebski) [inline packages/reassure-cli/src/commands/measure.ts:28]: Pls store `creationDate` as both date and time in ISO format. Do not separate it into `date` and `time` fields.

File: packages/reassure-compare/src/utils/format.ts [hunk 2]
```diff
@@ -99,8 +99,8 @@ function getRenderCountSymbols(entry: CompareEntry) {
 }
 
 export function formatMetadata(metadata?: PerformanceMetadata) {
-  if (metadata?.branch && metadata?.commitHash) {
-    return `${metadata.branch} (${metadata.commitHash})`;
+  if (metadata?.branch && metadata?.commitHash && metadata.creationDate) {
```
Comment (mdjastrzebski) [inline packages/reassure-compare/src/utils/format.ts:102]: Creation date is optional, older files will not contain it, therefor you should modify the code to support formatting in both cases with and without creation date

File: packages/reassure-compare/src/utils/validate.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ import { hasDuplicateValues } from './array';
 export const performanceMetadataSchema = z.object({
   branch: z.string().optional(),
   commitHash: z.string().optional(),
+  creationDate: z.string().optional(),
```
Comment (mdjastrzebski) [inline packages/reassure-compare/src/utils/validate.ts:9]: Pls use `z.string().datetime().optional()` to validate also the format of the date string (https://github.com/colinhacks/zod/releases/tag/v3.20)

**Comment by mdjastrzebski** [comment]
@josemak25 `main` branch is gathering results using Reassure in version before this code change, hence it will not have capture date before this PR is merged to `main`. After it is merged to `main` future baseline checks will run using the version of the code that captures the date.

File: packages/reassure-compare/src/utils/format.ts [hunk 4]
```diff
@@ -100,7 +100,7 @@ function getRenderCountSymbols(entry: CompareEntry) {
 
 export function formatMetadata(metadata?: PerformanceMetadata) {
   if (metadata?.branch && metadata?.commitHash) {
```
Comment (mdjastrzebski) [inline packages/reassure-compare/src/utils/format.ts:102]: Formatting branch + commit hash should be independent of formatting date:

```
function formatCommitMetadata() {
  if (metadata?.branch && metadata?.commitHash) {
    return `${metadata.branch} (${metadata.commitHash})
  }

 return metadata?.branch || metadata?.commitHash || '(unknown)';
}

function formatMetadata() {
  let result = formatCommitMetadata();
  if (metadata.creationDate) {
    result += ` - ${metadata.creationDate}`
  }

  return result;
}
```

File: .changeset/pretty-weeks-search.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+---
+'@callstack/reassure-compare': minor
+'@callstack/reassure-cli': minor
+'reassure': minor
+'@callstack/reassure-measure': minor
+---
+
+add data and time to measurement file
```
Comment (mdjastrzebski) [inline .changeset/pretty-weeks-search.md:8]:
```suggestion
feat: capture data and time of measurements
```

File: packages/reassure-compare/src/utils/format.ts (hunk 4 above)
Comment (josemak25) [reply packages/reassure-compare/src/utils/format.ts:102]: > Formatting branch + commit hash should be independent of formatting date:
> 
> ```
> function formatCommitMetadata() {
>   if (metadata?.branch && metadata?.commitHash) {
>     return `${metadata.branch} (${metadata.commitHash})
>   }
> 
>  return metadata?.branch || metadata?.commitHash || '(unknown)';
> }
> 
> function formatMetadata() {
>   let result = formatCommitMetadata();
>   if (metadata.creationDate) {
>     result += ` - ${metadata.creationDate}`
>   }
> 
>   return result;
> }
> ```

Alright I can resolve this

**Review by mdjastrzebski — APPROVED** [review]

**Pull request merged**